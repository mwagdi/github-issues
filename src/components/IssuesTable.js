import React from 'react';
import { connect } from 'react-redux';

import IssuesTableItem from './IssuesTableItem';
import { getIssues } from '../store/actions';

const IssuesTable = ({issues,getIssues}) => (
    <div className="issues-table flex-container flex-column">
        <div className="issues-table__header">
            <div className="flex-container">
                <a
                onClick={() => getIssues(1,"open")}
                className="get-issues open-get bold">Open</a>
                <a
                onClick={() => getIssues(1,"closed")}
                className="get-issues closed-get bold">Closed</a>
            </div>
        </div>

        <div className="issues-table__body">
            {issues.map((issue,i) => (
                <IssuesTableItem key={i} issue={issue} />
            ))}
        </div>
    </div>
);

export default connect(null,{ getIssues })(IssuesTable);