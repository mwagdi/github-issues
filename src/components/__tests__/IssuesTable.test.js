import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import IssuesTable from '../IssuesTable';

test('IssuesTable renders correctly', () => {
    const store = configureMockStore([])();
    const wrapper = shallow(<IssuesTable store={store} issues={[]} />).dive();
    expect(toJSON(wrapper)).toMatchSnapshot();
})
