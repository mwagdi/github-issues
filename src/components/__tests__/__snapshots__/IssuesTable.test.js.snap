// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesTable renders correctly 1`] = `
<div
  className="issues-table flex-container flex-column"
>
  <div
    className="issues-table__header"
  >
    <div
      className="flex-container"
    >
      <a
        className="get-issues open-get bold"
        onClick={[Function]}
      >
        Open
      </a>
      <a
        className="get-issues closed-get bold"
        onClick={[Function]}
      >
        Closed
      </a>
    </div>
  </div>
  <div
    className="issues-table__body"
  />
</div>
`;
